File: exchange/Order.ts
import { Order as CCTXOrder } from "ccxt";
import { Decimal } from "decimal.js";
export default class Order {
  id: string;
  symbol: string;
  side: string;
  type: string;
  timeInForce: string;
  amountRequested: Decimal;
  amountFilled: Decimal;
  avgFillPrice: Decimal;
  fee: Decimal;
  feeAsset: string;
  status: string;
  timestamp: number;
  /**
   *
   */
  constructor(cctxOrder: CCTXOrder) {
    this.id = cctxOrder.id.toString();
    this.symbol = cctxOrder.symbol;
    this.side = cctxOrder.side?.toString() ?? "";
    this.type = cctxOrder.type?.toString() ?? "";
    this.timeInForce = cctxOrder.timeInForce?.toString() ?? "";
    this.amountRequested = new Decimal(cctxOrder.amount);
    this.amountFilled = new Decimal(cctxOrder.filled);
    this.avgFillPrice = new Decimal(cctxOrder.average ?? 0);
    this.fee = new Decimal(Number(cctxOrder.fee?.cost));
    this.feeAsset = cctxOrder.fee?.currency?.toString() ?? "";
    this.status = cctxOrder.status?.toString() ?? ""; // 'filled', 'partially_filled', 'expired'
    this.timestamp = cctxOrder.timestamp;
  }
}

File: exchange/order.py
from decimal import Decimal


class Order:
    id: str
    symbol: str
    side: str
    type: str
    time_in_force: str
    amount_requested: Decimal
    amount_filled: Decimal
    avg_fill_price: Decimal
    fee: Decimal
    fee_asset: str
    status: str
    timestamp: int

    def __init__(self, ccxt_order: dict):
        self.id = str(ccxt_order.get("id", ""))
        self.symbol = ccxt_order.get("symbol", "")
        self.side = str(ccxt_order.get("side", ""))
        self.type = str(ccxt_order.get("type", ""))
        self.time_in_force = str(ccxt_order.get("timeInForce", ""))
        self.amount_requested = Decimal(str(ccxt_order.get("amount", 0)))
        self.amount_filled = Decimal(str(ccxt_order.get("filled", 0)))
        self.avg_fill_price = Decimal(str(ccxt_order.get("average", 0) or 0))
        self.timestamp = ccxt_order.get("timestamp", 0)
        self.status = str(ccxt_order.get("status", ""))

        fee = ccxt_order.get("fee") or {}
        self.fee = Decimal(str(fee.get("cost", 0) or 0))
        self.fee_asset = str(fee.get("currency", ""))

    def __repr__(self):
        return (
            f"Order(id={self.id}, symbol={self.symbol}, side={self.side}, "
            f"status={self.status}, filled={self.amount_filled}/{self.amount_requested} "
            f"@ {self.avg_fill_price})"
        )
